openapi-bal-service/src/main/java/io/ballerina/openapi/converter/diagnostic/ExceptionDiagnostic.java

```diff
@@ -33,10 +33,11 @@ public class ExceptionDiagnostic implements OpenAPIConverterDiagnostic {
     private final DiagnosticSeverity diagnosticSeverity;
     private final Location location;
 
-    public ExceptionDiagnostic(String code, String message, Location location, String... args) {
+    public ExceptionDiagnostic(String code, String message, Location location,
+                               DiagnosticSeverity severity, String... args) {
         this.code = code;
         this.message = String.format(message, args);
-        this.diagnosticSeverity = DiagnosticSeverity.ERROR;
+        this.diagnosticSeverity = severity;
```

Comment:
Is there any specific reason to make this `diagnosticSeverity` not an `ERROR`?

Reply:
Since we are setting diagnostic severity in the [DiagnosticMessages.java](https://github.com/ballerina-platform/openapi-tools/blob/master/openapi-bal-service/src/main/java/io/ballerina/openapi/converter/diagnostic/DiagnosticMessages.java) class, I find the assigning here as an over-ride of the information given in the original diagnostic message. I think its better to keep the severity at the message level since there is already a severity assigned. Or we can remove the seventies given in the message itself. Assigning a value and then overriding globally it might not be a good practice.

Reply:
Yeah, you are right! 😃 Here, we use ExceptionDiagnostic to populate an error when the OAS code generation cannot proceed (ex: parser issues, compilation issues, invalid input to CLI, and any Java exception that occurred). As far as I remember, we intentionally added this severity as an ERROR only for ExceptionDiagnostic to avoid misusing it, although the message includes its severity.

If we can guarantee that the given DiagnosticMessage uses ERROR as severity when it uses ExceptionDiagnostic, we can be safe by adding the severity from DiagnosticMessage.WDYT @aneeshafedo @NipunaRanasinghe ?

Reply:
Reverted the change